src/schema/schema_drop.c

```diff
@@ -182,10 +182,15 @@ __drop_tiered(WT_SESSION_IMPL *session, const char *uri, bool force, const char
     WT_TIERED *tiered;
     u_int i;
     const char *filename, *name;
-    bool exist, remove_files;
+    bool exist, remove_files, remove_shared;
 
     WT_RET(__wt_config_gets(session, cfg, "remove_files", &cval));
     remove_files = cval.val != 0;
+    WT_RET(__wt_config_gets(session, cfg, "remove_shared", &cval));
+    remove_shared = cval.val != 0;
+
+    if (!remove_files && remove_shared)
+        WT_RET(EINVAL);
```

Comment:
We probably should provide an error message here with `WT_RET_MSG`.

Reply:
Good suggestion, thanks!